hooks/queries/useProgramVersionQuery.ts

```diff
@@ -0,0 +1,29 @@
+import { getGovernanceProgramVersion } from '@solana/spl-governance'
+import { PublicKey } from '@solana/web3.js'
+import { useQuery } from '@tanstack/react-query'
+import useWalletStore from 'stores/useWalletStore'
+
+export const programVersionQueryKeys = {
+  byProgramId: (programId: PublicKey) => [
+    'programVersion',
+    programId.toString(),
+  ],
+}
+
+export function useProgramVersionByIdQuery(realmsProgramId?: PublicKey) {
+  // @asktree is unsure why we use this instead of `useConnection` (which has no corresponding provider in the BaseApp)
+  const { connection } = useWalletStore()
+  console.log('connection', connection)
+
+  const query = useQuery({
+    queryKey:
+      realmsProgramId && programVersionQueryKeys.byProgramId(realmsProgramId),
+    queryFn: () =>
+      getGovernanceProgramVersion(connection.current, realmsProgramId!),
+    enabled: realmsProgramId !== undefined,
```

Comment:
We can stop the queryFn from executing if the input is undefined, thus letting us make optional queries.